coordinator/src/services/translation/request/payload/factory.ts

```diff
@@ -0,0 +1,203 @@
+import Hapi from "@hapi/hapi"
+import {fhir, hl7V3} from "@models"
+import pino from "pino"
+import * as uuid from "uuid"
+import {
+  getMessageIdFromBundle,
+  getMessageIdFromClaim,
+  getMessageIdFromTask,
+  identifyMessageType
+} from "../../common"
+import {convertCancellation} from "../cancel/cancellation"
+import {convertDispenseClaim} from "../dispense/dispense-claim"
+import {convertDispenseNotification} from "../dispense/dispense-notification"
+import {translateReleaseRequest} from "../dispense/release"
+import {convertParentPrescription} from "../prescribe/parent-prescription"
+import {convertTaskToDispenseProposalReturn} from "../return/return"
+import {createSendMessagePayload} from "../send-message-payload"
+import {convertTaskToEtpWithdraw} from "../withdraw/withdraw"
+
+type BundleTranslationResult = hl7V3.ParentPrescriptionRoot
+  | hl7V3.CancellationRequestRoot
+  | hl7V3.DispenseNotificationRoot
+  | hl7V3.DispenseClaimRoot
+
+type ParametersTranslationResult = hl7V3.NominatedPrescriptionReleaseRequestWrapper
+  | hl7V3.PatientPrescriptionReleaseRequestWrapper
+
+type TaskTranslationResult = hl7V3.DispenseProposalReturnRoot | hl7V3.EtpWithdrawRoot
+
+type ClaimTranslationResult = hl7V3.DispenseClaimRoot
+
+export type PayloadContent = BundleTranslationResult
+  | TaskTranslationResult
+  | ParametersTranslationResult
+  | ClaimTranslationResult
+
+type Payload<T extends PayloadContent> = {
+  content: T,
+  interactionId: hl7V3.Hl7InteractionIdentifier
+}
+
+type FactoryInput = fhir.Bundle | fhir.Task | fhir.Parameters | fhir.Claim
+
+export abstract class PayloadFactory {
+  protected abstract create(
+    fhirResource: FactoryInput,
+    logger?: pino.Logger
+  ): Payload<PayloadContent>
+
+  protected abstract getPayloadId(fhirResource: FactoryInput): string
+
+  static forBundle(): PayloadFactory {
+    return new BundleTranslationResultFactory()
+  }
+
+  static forTask(): PayloadFactory {
+    return new TaskTranslationResultFactory()
+  }
+
+  static forParameters(): PayloadFactory {
+    return new ParametersTranslationResultFactory()
+  }
+
+  static forClaim(): PayloadFactory {
+    return new ClaimTranslationResultFactory()
+  }
+
+  public makeSendMessagePayload(
```

Comment:
in a similar vein to the comment above, `create` is used in a method called `makeSendMessagePayload`, so `make` might be a better word?